src/rust/engine/src/lib.rs

```diff
@@ -782,6 +785,35 @@ pub extern "C" fn materialize_directories(
   .into()
 }
 
+#[no_mangle]
+pub extern "C" fn setup_pantsd_logger(log_file_ptr: *const raw::c_char, level: u64) -> PyResult {
+  let path_str = unsafe { CStr::from_ptr(log_file_ptr).to_string_lossy().into_owned() };
+  let path = PathBuf::from(path_str);
+  LOGGER.set_pantsd_logger(path, level).into()
+}
+
+#[no_mangle]
+pub extern "C" fn setup_stderr_logger(level: u64) {
+  LOGGER
+    .set_stderr_logger(level)
+    .expect("Error setting up STDERR logger");
```

Comment:
Maybe return a `Result` here?

Reply:
Sometimes these functions get called before externs are initialised, so they will not be able to instantiate a `PyResult`